daffodil-runtime2/src/main/scala/org/apache/daffodil/runtime2/CodeGenerator.scala

```diff
@@ -198,3 +200,60 @@ class CodeGenerator(root: Root) extends DFDL.CodeGenerator {
   override def getDiagnostics: Seq[Diagnostic] = diagnostics
   override def isError: Boolean = errorStatus
 }
+
+/** Runs from "sbt compile" to keep all example generated code files up to date */
+object CodeGenerator {
+  // Update one set of example generated code files from an example schema
+  private def updateGeneratedCodeExample(schemaFile: os.Path, rootName: Option[String],
+                                         exampleCodeHeader: os.Path, exampleCodeFile: os.Path): Unit = {
+    // Generate code from the example schema file
+    val pf = Compiler().compileFile(schemaFile.toIO, rootName)
+    assert(!pf.isError, pf.getDiagnostics.map(_.getMessage()).mkString("\n"))
+    val cg = pf.forLanguage("c")
+    val rootNS = QName.refQNameFromExtendedSyntax(rootName.getOrElse("")).toOption
+    val tempDir = os.temp.dir(dir = null, prefix = "daffodil-runtime2-")
+    val codeDir = cg.generateCode(rootNS, tempDir.toString)
+    assert(!cg.isError, cg.getDiagnostics.map(_.getMessage()).mkString("\n"))
+
+    // Replace the example generated files with the newly generated files
+    val generatedCodeHeader = codeDir/"libruntime"/"generated_code.h"
+    val generatedCodeFile = codeDir/"libruntime"/"generated_code.c"
+    os.copy(generatedCodeHeader, exampleCodeHeader, replaceExisting = true, createFolders = true)
+    os.copy(generatedCodeFile, exampleCodeFile, replaceExisting = true, createFolders = true)
+
+    // Print the example generated files' names so "sbt 'show genExamples'" can list them
+    System.out.println(exampleCodeHeader)
+    System.out.println(exampleCodeFile)
+
+    // tempDir should be removed automatically after main exits; this is just in case
+    os.remove.all(tempDir)
+  }
+
+  // Make sure "sbt compile" calls this main method
+  def main(args: Array[String]): Unit = {
```

Comment:
I'm not sure this is a big deal, but if anyone ever runts this jar using java -jar it's probably going to fail, since it can only be run in the daffodil root dir. Like, one might expect this to be an alternative to using the CLI, but it's really a test only thing. Maybe it should go in src/test/scala so it's not distributed? Or maybe it doesn't really matter.

Reply:
I don't think it's a big deal that this main method works only in the daffodil root directory or the daffodil-runtime2 directory (see the if expression on line 235).  The only two times I expect anyone to run this main method is from "sbt compile" (where the daffodil root directory is the current directory) or from their IDE as a runtime configuration (where the daffodil-runtime2 directory is the current directory).